pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -369,6 +369,11 @@ if [[ -n $flake ]]; then
     else
         flakeAttr="nixosConfigurations.\"$flakeAttr\""
     fi
+
+    # Follow symlink if local path
+    if [[ -e $flake ]] then
+        flake="$(readlink -f $flake)"
+    fi
```

Comment:
Shouldn't this be combined with line 352?

Reply:
We could set `flake="/etc/nixos"` there, then single `readline` would suffice here. Though ideally, we would fix Nix to support building flakes using symlink paths. Maybe it would deserve a TODO comment linking https://github.com/NixOS/nix/issues/8013 here so that we know we can drop this once it is resolved.

Reply:
I think it should even be `flake="/etc/nixos/flake.nix"` because having `/etc/nixos/flake.nix` be a symlink to `/some/repo/flake.nix` works (thanks to the call to dirname after readlink).

Reply:
I also thought it wouldn't make sense wouldn't work but the current behavior of `nixos-rebuild switch` accepts it. The call to dirname after readlink resolves to the real folder containing the flake.nix.

I'll let you be the judge on this one but I would find it weird for `nixos-rebuild switch` to behave differently from `nixos-rebuild switch --flake /etc/nixos/flake.nix`

Reply:
Passing `/etc/nixos/flake.nix` to `--flake` argument is just wrong.  `--flake` expects a flake “URI”, which can be a path to a directory containing `flake.nix`. Passing it a file is not expected – though, it may still work with a warning “path '…/flake.nix' does not contain a 'flake.nix', searching up”.

Reply:
Fair enough. In that case I should revert line 352 or this PR will introduce a breaking change.